src/Validator/Metadata/Resource/Factory/ParameterValidationResourceMetadataCollectionFactory.php

```diff
@@ -190,7 +191,18 @@ private function addSchemaValidation(Parameter $parameter, ?array $schema = null
         }
 
         if (isset($schema['type']) && 'array' === $schema['type']) {
-            $assertions[] = new Type(type: 'array');
+            $assertions[] = new Type(type: $schema['type']);
+        }
+
+        // Allow null in case of optional parameter
+        if (isset($schema['type']) && 'null' === $schema['type']) {
+            // Header null values may be sent as array through Symfony
+            $assertions[] = new Expression(expression: 'value in [null, "null", [null], ["null"]]');
```

Comment:
This should just be `Assert\IsNull`, values could be transformed by a ParameterProvider to change "null" to `null`. Though, this is quite specific, I don't see any use case and it'll probably lead to unwanted behaviors. I'd suggest that we leave that out and we don't add any validation on that type.